etna/transforms/missing_values/imputation.py

```diff
@@ -2,25 +2,67 @@
 from typing import Dict
 from typing import List
 from typing import Optional
-from typing import cast
+from typing import Sequence
 
+import bottleneck as bn
 import numpy as np
 import pandas as pd
+from sklearn.impute import SimpleImputer
 
 from etna.distributions import BaseDistribution
 from etna.distributions import CategoricalDistribution
 from etna.distributions import IntDistribution
+from etna.transforms import LagTransform
+from etna.transforms import MeanTransform
 from etna.transforms.base import ReversibleTransform
 from etna.transforms.utils import check_new_segments
 
 
+class SimpleImputerSubsegment(SimpleImputer):
+    def __init__(
+        self,
+        missing_values=np.nan,
+        strategy="mean",
+        fill_value=None,
+        verbose=0,
+        copy=True,
+        add_indicator=False,
+    ):
+        super().__init__(
+            missing_values=missing_values,
+            strategy=strategy,
+            fill_value=fill_value,
+            verbose=verbose,
+            copy=copy,
+            add_indicator=add_indicator,
+        )
+        self._segment_to_index: Optional[Dict[str, int]] = None
+
+    def fit(self, X, y=None):  # noqa: N803
+        X.sort_index(axis=1, inplace=True)
+        segments = sorted(X.columns.get_level_values("segment").unique())
+        self._segment_to_index = {segment: i for i, segment in enumerate(segments)}
+        super().fit(X.values)
+
+    def transform(self, X):  # noqa: N803
+        X.sort_index(axis=1, inplace=True)
+        segments = X.columns.get_level_values("segment").unique()
+        old_statistics = self.statistics_.copy()
+        self.n_features_in_ = len(segments)
```

Comment:
What happens if during `super().transform(X.values)` it fails for some reason. Do we have incorrect `        self.n_features_in_` in this case?

Reply:
Yes, fixed it

Reply:
Do we assume that there couldn't be new segments in `segments`? If so, then code will fail on `self.statistics_ = ...` before `try/finally` block.

Reply:
If we use this class only inside `TimeSeriesImputerTransform` we do